frontera/contrib/scrapy/schedulers/frontier.py

```diff
@@ -124,11 +134,13 @@ def process_exception(self, request, exception, spider):
     def open(self, spider):
         log.msg('Starting frontier', log.INFO)
         if not self.frontier.manager.auto_start:
-            self.frontier.start()
+            scrapy_kwargs = {'spider': spider}
+            self.frontier.start(**scrapy_kwargs)
```

Comment:
Regarding first two points, the general way to pass all the request information is to use request.meta dictionary in spider, and put all the slot and other metadata there and later access them in backend. Same for headers/method. You can even put additional fields in Request object, or customize Request/Response models in Frontera.

Regarding third one, frontier parameters. May be it's better to use Frontera settings file for this? Or there is something requiring to do this directly from spider code?